src/main/java/org/adridadou/ethereum/propeller/rpc/Web3JFacade.java

```diff
@@ -62,6 +61,27 @@ EthData constantCall(final EthAccount account, final EthAddress address, final E
         }
     }
 
+    List<Log> loggingCall(SolidityEvent eventDefiniton, final EthAddress address, final String... optionalTopics) {
+        EthFilter ethFilter = new EthFilter(DefaultBlockParameterName.EARLIEST, DefaultBlockParameterName.LATEST, address.withLeading0x());
+
+        ethFilter.addSingleTopic(EventEncoder.buildEventSignature(buildMethodSignature(eventDefiniton.getDescription().getName(), eventDefiniton.getDescription().getInputs())));
+        ethFilter.addOptionalTopics(optionalTopics);
+
+        List<Log> list = new ArrayList();
+        this.web3j.ethLogFlowable(ethFilter).subscribe(log -> list.add(log)).dispose();
+        return list;
+    }
+
+    private String buildMethodSignature(String methodName, List<AbiParam> parameters) {
```

Comment:
Instead of writing the method here, I would go and change AbiEntry.
Instead of the implementations for signature and signatureLong, we should have something like that:

```
public EthData signature() {
        //Constructor has no signature has it is an anonyous function
        if ("constructor".equals(type)) {
            return EthData.empty();
        }

        return EthData.of(Arrays.copyOfRange(signatureLong().data, 0, 4));
    }

    public EthData signatureLong() {
        return EthData.of(sha3((signatureClear()).getBytes()));
    }

    public String signatureClear() {
        String params = getInputs().stream().map(AbiParam::getType).collect(Collectors.joining(","));
        return getName() + "(" + params + ")";
    }
```

Reply:
and then you can simply use signatureClear instead of your own implementation

Reply:
Think this is resolved, removed the custom code. SignatureLong actually works fine but I just wasn't aware of the method being available!